### PR TITLE
Add Titles!

### PR DESCRIPTION
**Pull request by LucGamesYT**

### CODE REVIEW

File: gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java [hunk 1]
```diff
@@ -319,4 +319,30 @@
      */
     boolean getFlying();
 
+    /**
+     *
```
Comment (geNAZt) [inline gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java:324]: Documentation pls

File: gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java [hunk 2]
```diff
@@ -319,4 +319,30 @@
...
+     * @param title
+     * @param subtitle
+     * @param fadein
+     * @param duration
+     * @param fadeout
+     */
+
+    void sendTitle( String title, String subtitle, int fadein, int duration, int fadeout );
```
Comment (geNAZt) [inline gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java]: All gomint-api is not based on ticks. Please use long, TimeUnit when you need duration times (in this case for fadeIn, duration and fadeOut)

File: gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java [hunk 3]
```diff
@@ -319,4 +319,30 @@
...
+
+    /**
+     *
```
Comment (geNAZt) [inline gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java]: Documentation should include what default times are used

File: gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java [hunk 4]
```diff
@@ -319,4 +319,30 @@
...
+     * @param title
+     */
+
+    void sendTitle( String title );
+
+    /**
+     *
```
Comment (geNAZt) [inline gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java:347]: Same here (default times)

File: gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package io.gomint.server.network.packet;
+
+import io.gomint.jraknet.PacketBuffer;
+import io.gomint.server.network.Protocol;
+import lombok.Getter;
+import lombok.Setter;
+
+public class PacketSetTitle extends Packet {
+
+    public static final int TYPE_CLEAR = 0;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java]: Type declarations should be done in a enum instead

File: gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    public static final int TYPE_RESET = 1;
+    public static final int TYPE_TITLE = 2;
+    public static final int TYPE_SUBTITLE = 3;
+    public static final int TYPE_ACTION_BAR = 4;
+    public static final int TYPE_ANIMATION_TIMES = 5;
+
+    @Getter@Setter
+    public int type;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java]: No public access to variables pls

File: gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+public class PacketSetTitle extends Packet {
+
+    public static final int TYPE_CLEAR = 0;
+    public static final int TYPE_RESET = 1;
+    public static final int TYPE_TITLE = 2;
+    public static final int TYPE_SUBTITLE = 3;
+    public static final int TYPE_ACTION_BAR = 4;
+    public static final int TYPE_ANIMATION_TIMES = 5;
+
+    @Getter@Setter
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java]: Use @Data above the Class definition (see other packets)

**Review by geNAZt — CHANGES_REQUESTED** [review]

**Review by Janmm14 — CHANGES_REQUESTED** [review]
no normal text documentation below any @ -tags pls

File: gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java [hunk 8]
```diff
@@ -319,4 +320,40 @@
...
+     * Default time for fadin and duration is 1 second.
+     */
+
+    void sendTitle( String title );
+
+    /**
+     *
+     * Send a title with title and subtitle.
+     *
+     * @param title
```
Comment (Janmm14) [inline gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java]:
```
@param title Big text displayed in the middle of the screen
@param Subtitle Smaller big text displayed below the title text
```

File: gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
+package io.gomint.server.network.packet;
+
+import io.gomint.jraknet.PacketBuffer;
+import io.gomint.server.network.Protocol;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+
+@Data
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java]: @Data means you do not need `@Getter` and `@Setter` anymore on each field

File: gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        TYPE_RESET( 1 ),
+        TYPE_TITLE( 2 ),
+        TYPE_SUBTITLE( 3 ),
+        TYPE_ACTION_BAR( 4 ),
+        TYPE_ANIMATION_TIMES( 5 );
+
+        @Getter
+        private final int id;
+
+        TitleType( int id ) {
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/network/packet/PacketSetTitle.java]: maybe add `@RequiredArgsConstructor` instead (if it works on enums)

**Pull request merged**